mysql-test/suite/rocksdb/t/use_direct_reads_writes.test

```diff
@@ -5,43 +5,37 @@
 # caused an assertion in RocksDB.  Now it should not be allowed and the
 # server will not start with that configuration
 
-# Write file to make mysql-test-run.pl expect the "crash", but don't restart
-# the server until it is told to
---let $_server_id= `SELECT @@server_id`
---let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.$_server_id.expect
---exec echo "wait" >$_expect_file_name
-shutdown_server 10;
+--let SEARCH_FILE=$MYSQLTEST_VARDIR/tmp/rocksdb.use_direct_reads_writes.err
 
-# Clear the log
---exec echo "" >$MYSQLTEST_VARDIR/log/mysqld.1.err
+# Expect server crash on startup with invalid combination of arguments
+--source include/expect_crash.inc
+shutdown_server 10;
 
 # Attempt to restart the server with invalid options
---exec echo "restart:--rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1" >$_expect_file_name
---sleep 0.1  # Wait 100ms - that is how long the sleep is in check_expected_crash_and_restart
---exec echo "restart:" >$_expect_file_name
-
-# Cleanup
---enable_reconnect
---source include/wait_until_connected_again.inc
---disable_reconnect
+--let $restart_parameters=restart:--log-error=$SEARCH_FILE --rocksdb_use_direct_reads=1 --rocksdb_allow_mmap_reads=1
+--replace_result $SEARCH_FILE LOG_FILE
+--source include/start_mysqld.inc
```

Comment:
There is no _exact_ example that matches this case with a plugin load needed. For some reason the changes I pushed run correctly locally, but fail across 100% of our jenkins farm:
> 17:26:54 rocksdb.use_direct_reads_writes          w3 [ fail ]
> 17:26:54         Test ended at 2017-08-15 21:26:54
> 17:26:54 
> 17:26:54 CURRENT_TEST: rocksdb.use_direct_reads_writes
> 17:26:54 mysqld-debug: Can't change dir to '/var/lib/mysql/' (Errcode: 2 - No such file or directory)
> 17:26:54 2017-08-15T21:26:50.065769Z 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
> 17:26:54 2017-08-15T21:26:50.065865Z 0 [Note] --secure-file-priv is set to NULL. Operations related to importing and exporting data are disabled
> 17:26:54 2017-08-15T21:26:50.065897Z 0 [Note] /mnt/workspace/mysql-5.7-param/BUILD_TYPE/debug/Host/ubuntu-trusty-64bit/build/sql/mysqld-debug (mysqld 5.7.18-16-debug) starting as process 17851 ...
> 17:26:54 2017-08-15T21:26:50.065941Z 0 [ERROR] Can't find error-message file '/usr/share/mysql/errmsg.sys'. Check error-message file location and 'lc-messages-dir' configuration directive.
> 17:26:54 2017-08-15T21:26:50.067389Z 0 [Warning] Can't create test file /var/lib/mysql/os-trusty-x64-2030.lower-test
> 17:26:54 2017-08-15T21:26:50.067408Z 0 [Warning] Can't create test file /var/lib/mysql/os-trusty-x64-2030.lower-test
> 17:26:54 2017-08-15T21:26:50.067811Z 0 [ERROR] failed to set datadir to /var/lib/mysql/
> 17:26:54 2017-08-15T21:26:50.067823Z 0 [ERROR] Aborting

I suspect this might be some odd difference between in-source and out-of-source builds. 
I have tried other ways of invoking --exec $MYSQLD based on numerous existing tests but every combination seems to run into one issue or another. So I'm about out of ideas here how this is supposed to work and can't find anything reasonably close in any documentation or existing suites.

Please take a look at the pushed changes and see if there is something obvious that I am missing here.